server/room-battle.js

```diff
@@ -502,16 +508,26 @@ class RoomBattle {
 
 		if (options.p1) this.addPlayer(options.p1, 'p1', options.p1team, true);
 		if (options.p2) this.addPlayer(options.p2, 'p2', options.p2team, true);
+		if (options.p3) this.addPlayer(options.p3, 'p3', options.p3team, true);
+		if (options.p4) this.addPlayer(options.p4, 'p4', options.p4team, true);
 	}
 
 	checkActive() {
 		let active = true;
 		if (this.ended || !this.started) {
 			active = false;
-		} else if (!this.p1 || !this.p1.active) {
```

Comment:
This part is pretty unnecessary.

Reply:
So just remove the entire check?

Reply:
No, I mean `else { if }` means the exact same thing as `else if`, so there was no reason for you to change it.

Reply:
Like it should have just been

```js
} else if (p1) {
} else if (p2) {
} else if (fourPlayers) {
  if (p3) {
  } else if (p4) {
  }
}
```

Reply:
Oh, yeah, no idea why I did it this way.